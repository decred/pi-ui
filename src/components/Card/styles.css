.card {
  background: var(--color-white);
}
