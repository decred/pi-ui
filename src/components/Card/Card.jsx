import PropTypes from "prop-types";
import React from "react";
import { classNames } from "../../utils";
import styles from "./styles.module.css";

const Card = ({
  children,
  className,
  style,
  paddingSize,
  marker,
  markerColor,
  ...props
}) => (
  <div
    className={classNames(
      styles.card,
      styles[`padding-${paddingSize}`],
      className
    )}
    style={style}
    {...props}>
    {marker && (
      <div
        className={styles.marker}
        style={{
          borderLeftColor: markerColor,
          borderTopColor: markerColor,
        }}
      />
    )}
    {children}
  </div>
);

Card.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  paddingSize: PropTypes.oneOf(["small", "medium", "large"]),
  marker: PropTypes.bool,
  markerColor: PropTypes.string,
  style: PropTypes.object,
};

Card.defaultProps = {
  markerColor: "var(--color-yellow)",
};

export default Card;
