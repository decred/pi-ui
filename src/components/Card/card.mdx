---
name: Card
route: /card
---

import { Playground, PropsTable } from "docz";
import Card from "./Card.jsx";

# Card

## Properties

<PropsTable of={Card} />

## Usage

<Playground  style={{ background: "#c4cbd2" }}>
     <Card>I am a Card</Card>
     <br />
     <Card style={{ padding: "10px" }}>I am a Card with padding</Card>
</Playground>
