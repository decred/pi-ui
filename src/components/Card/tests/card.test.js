import React from "react";
import Card from "../Card";
import { create } from "react-test-renderer";

describe("Card component", () => {
  test("Matches the snapshot", () => {
    const card = create(<Card>test</Card>);
    expect(card.toJSON()).toMatchSnapshot();
  });
});
