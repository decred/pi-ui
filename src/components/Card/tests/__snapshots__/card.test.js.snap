// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Matches the snapshot 1`] = `
<div
  className="card padding-undefined"
>
  test
</div>
`;
